app/src/main/java/rightmesh/left/io/gpstracker/LocationTracker.java

```diff
@@ -0,0 +1,198 @@
+package rightmesh.left.io.gpstracker;
+
+import android.Manifest;
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.app.PendingIntent;
+import android.arch.lifecycle.Lifecycle;
+import android.arch.lifecycle.LifecycleObserver;
+import android.arch.lifecycle.OnLifecycleEvent;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.location.Location;
+import android.location.LocationManager;
+import android.support.v4.app.ActivityCompat;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationCallback;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.location.LocationSettingsRequest;
+import com.google.android.gms.location.SettingsClient;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+
+/**
+ * Using FusedLocationProviderClient to get location updates.
+ */
+public class LocationTracker implements LifecycleObserver {
+    private FusedLocationProviderClient fusedLocationProviderClient;
+    private Activity activity;
+
+    private int interval = 1000;
+    private int fastestInterval = 500;
+    private int priority = LocationRequest.PRIORITY_HIGH_ACCURACY;
+
+    private LocationRequest locationRequest;
+
+    private LocationCallback locationCallback;
+
+    /**
+     * LocationTracker contructor.
+     *
+     * @param activity  current activity
+     * @param lifecycle get by getLifecycle() in Activity, Fragment.
+     *                  Used to observe lifecycle of Activity or Fragment
+     */
+    public LocationTracker(Activity activity, Lifecycle lifecycle) {
+        this.activity = activity;
+        this.fusedLocationProviderClient = LocationServices
+                .getFusedLocationProviderClient(activity);
+        lifecycle.addObserver(this);
+    }
+
+    /**
+     * Set up the preferred the rate in millisecond to receive updates.
+     *
+     * @param interval in milliseconds
+     * @return {@link LocationTracker}
```

Comment:
It is helpful to describe the content that is being returned, beyond just the type. For example this tag you could say

`@return this {@link LocationTracker} instance`.

Elsewhere maybe `@return {@link Type} representing whatever, of <foo> in <this edge case>`.